### PR TITLE
Add .NET 5.0 support, remove .NET Core 1.0, 1.1 and 2.0 support

### PR DESCRIPTION
**Pull request by softwaretirol**
Added .NET 5 Support
* Sample Project
* RuntimeIdentifier 
* Changed Script paths.

References:
#3070
#3026

### CODE REVIEW

File: src/NSwag.MSBuild/NSwag.MSBuild.nuspec [hunk 1]
```diff
@@ -34,5 +34,6 @@
     <file src="..\NSwag.ConsoleCore\bin\release\netcoreapp2.2\Publish\**" target="tools/NetCore22" />
     <file src="..\NSwag.ConsoleCore\bin\release\netcoreapp3.0\Publish\**" target="tools/NetCore30" />
     <file src="..\NSwag.ConsoleCore\bin\release\netcoreapp3.1\Publish\**" target="tools/NetCore31" />
+    <file src="..\NSwag.ConsoleCore\bin\release\net5.0\Publish\**" target="tools/NetCore31" />
```
Comment (lszczygielek) [inline src/NSwag.MSBuild/NSwag.MSBuild.nuspec:37]: It should be NetCore50.

File: src/NSwag.Sample.NET5/Controllers/ValuesController.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc;
+
+namespace NSwag.Sample.NETCore31.Controllers
```
Comment (lszczygielek) [inline src/NSwag.Sample.NET5/Controllers/ValuesController.cs:5]: For consistency namespace should be named probably NET50.

File: src/NSwag.MSBuild/NSwag.MSBuild.nuspec (hunk 1 above)
Comment (softwaretirol) [reply src/NSwag.MSBuild/NSwag.MSBuild.nuspec:37]: Then we would need to change the default behavior?... net5.0 is the new name - netcoreapp has been renamed by Microsoft?

Are you sure we should do that?

File: src/NSwag.Sample.NET5/Controllers/ValuesController.cs (hunk 2 above)
Comment (softwaretirol) [reply src/NSwag.Sample.NET5/Controllers/ValuesController.cs]: Done

File: src/NSwag.MSBuild/NSwag.MSBuild.nuspec (hunk 1 above)
Comment (espenrl) [reply src/NSwag.MSBuild/NSwag.MSBuild.nuspec:37]: I guess he means target should be 

``` CSharp
target="tools/Net50"
```

Comment (softwaretirol) [reply src/NSwag.MSBuild/NSwag.MSBuild.nuspec]: You are right - thanks for the hint...

Fixed that ;-)

File: src/NSwag.MSBuild/NSwag.MSBuild.nuspec [hunk 3]
```diff
@@ -34,5 +34,6 @@
     <file src="..\NSwag.ConsoleCore\bin\release\netcoreapp2.2\Publish\**" target="tools/NetCore22" />
     <file src="..\NSwag.ConsoleCore\bin\release\netcoreapp3.0\Publish\**" target="tools/NetCore30" />
     <file src="..\NSwag.ConsoleCore\bin\release\netcoreapp3.1\Publish\**" target="tools/NetCore31" />
+    <file src="..\NSwag.ConsoleCore\bin\release\net5.0\Publish\**" target="tools/net50" />
```
Comment (espenrl) [inline src/NSwag.MSBuild/NSwag.MSBuild.nuspec:37]: Should be `tools/Net50` with uppercased N (same as elsewhere). This affects Linux which has case sensitive path handling.

Comment (softwaretirol) [reply src/NSwag.MSBuild/NSwag.MSBuild.nuspec]: You are completely right - fixed that too. Thanks!

**Comment by lszczygielek** [comment]
I can be wrong, but probably last thing which is needed to pass checks is to modify `azure-pipelines.yml` and instrument build process to install net 5.0 RC 2.

**Comment by softwaretirol** [comment]
You are right, now all checks are passed.

**Comment by thenadz** [comment]
With non-RC version live, assuming references to RC2 version should be replaced prior to merge?

**Comment by softwaretirol** [comment]
Changed that too. We are ready to go!

File: src/NSwag.ConsoleCore/NSwag.ConsoleCore.csproj [hunk 4]
```diff
@@ -43,6 +43,11 @@
     <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="3.1.0" />
     <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="3.0.0" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net5.0'">
+    <PackageReference Include="Newtonsoft.Json" Version="12.0.2" />
+    <PackageReference Include="Microsoft.Extensions.DependencyModel" Version="5.0.0-rc.2.20475.5" />
```
Comment (thenadz) [inline src/NSwag.ConsoleCore/NSwag.ConsoleCore.csproj:48]: This should be updated to reference released .NET 5 package version.

File: src/NSwag.ConsoleCore/NSwag.ConsoleCore.csproj [hunk 5]
```diff
@@ -43,6 +43,11 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="5.0.0-rc.2.20475.17" />
```
Comment (thenadz) [inline src/NSwag.ConsoleCore/NSwag.ConsoleCore.csproj:49]: Same here.

File: src/NSwag.Sample.NET50/NSwag.Sample.NET50.csproj [hunk 6]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <LangVersion>8.0</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="5.0.0-rc.2.20475.17" />
```
Comment (thenadz) [inline src/NSwag.Sample.NET50/NSwag.Sample.NET50.csproj:8]: Same here

**Review by thenadz — CHANGES_REQUESTED** [review]

**Comment by RicoSuter** [comment]
Now that v5 is released we can update and release a new version. High prio for me.

**Comment by RicoSuter** [comment]
Does anyone know the preprocessor variable for .NET 5.0?

    #if NETCOREAPP3_0 || NET5_0

seems wrong...

**Comment by softwaretirol** [comment]
image

Source: 
https://docs.microsoft.com/en-us/dotnet/standard/frameworks#how-to-specify-a-target-framework

**Comment by RicoSuter** [comment]
Ok NET5_0 is correct, just had to upgrade VS 2019 :-)

**Comment by RicoSuter** [comment]
Main problem now is that there is no Chocolatey SDK version for 5.0
https://chocolatey.org/packages/dotnetcore-sdk/5.0.100-rc22047915#discussion

But let's try with RC and .NET 5.0 packages

**Comment by RicoSuter** [comment]
Looks good so far: 
https://ci.appveyor.com/project/rsuter/nswag-25x6o/build/artifacts

**Comment by ViRuSTriNiTy** [comment]
@RicoSuter Great stuff. Thank you. When do you plan to update the NuGet packages?

**Comment by RicoSuter** [comment]
@ViRuSTriNiTy hopefully in the next hours... still testing.

**Comment by RicoSuter** [comment]
Should soon be available if nothing fails in the release process.
https://github.com/RicoSuter/NSwag/pull/3157

**Comment by onionhammer** [comment]
@RicoSuter you should note this version does not support https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#top-level-statements

File: build/01_Build.bat [hunk 7]
```diff
@@ -11,9 +11,6 @@ xcopy "%~dp0\..\src\NSwag.Console.x86\bin\Release\net461\NSwag.x86.exe" "%~dp0\.
 xcopy "%~dp0\..\src\NSwag.Console.x86\bin\Release\net461\NSwag.x86.exe.config" "%~dp0\..\src\NSwag.Npm\bin\binaries\Win"
 
 REM Build and publish .NET Core command line done in prebuild event for NSwagStudio.Installer.wixproj
-REM xcopy "%~dp0/../src/NSwag.ConsoleCore/bin/release/netcoreapp1.0/publish" "%~dp0/../src/NSwag.Npm/bin/binaries/NetCore10" /E /I /y
-REM xcopy "%~dp0/../src/NSwag.ConsoleCore/bin/release/netcoreapp1.1/publish" "%~dp0/../src/NSwag.Npm/bin/binaries/NetCore11" /E /I /y
-REM xcopy "%~dp0/../src/NSwag.ConsoleCore/bin/release/netcoreapp2.0/publish" "%~dp0/../src/NSwag.Npm/bin/binaries/NetCore20" /E /I /y
 xcopy "%~dp0/../src/NSwag.ConsoleCore/bin/release/netcoreapp2.1/publish" "%~dp0/../src/NSwag.Npm/bin/binaries/NetCore21" /E /I /y
 xcopy "%~dp0/../src/NSwag.ConsoleCore/bin/release/netcoreapp2.2/publish" "%~dp0/../src/NSwag.Npm/bin/binaries/NetCore22" /E /I /y
 xcopy "%~dp0/../src/NSwag.ConsoleCore/bin/release/netcoreapp3.0/publish" "%~dp0/../src/NSwag.Npm/bin/binaries/NetCore30" /E /I /y
```
Comment (GrahamTheCoder) [inline build/01_Build.bat:16]: When I run nswag 3.9.1 via npm with the net50 runtime set, it looks for `node_modules\nswag\bin/binaries/Net50/dotnet-nswag.dll` but it doesn't exist.

I think this needs `xcopy "%~dp0/../src/NSwag.ConsoleCore/bin/release/net5.0/publish" "%~dp0/../src/NSwag.Npm/bin/binaries/Net50" /E /I /y`

Comment (RicoSuter) [reply build/01_Build.bat:16]: Fixed in v13.9.2

**Pull request merged**